stl/inc/xlocmon

```diff
@@ -8,6 +8,7 @@
 #define _XLOCMON_
 #include <yvals_core.h>
 #if _STL_COMPILER_PREPROCESSOR
+#include <iterator>

```

Comment:
Including `<iterator>` is unnecessary in `<xlocmon>` and `<xloctime>`, and it's slightly inconsistent with the existing code that appears to treat `<xlocnum>` as central. However, I'm fine with this change. (STL headers are wildly inconsistent about assuming transitive inclusion, although we try to be very clean in our tests.)